pkg/openSearch/openSearchClient/config/config.go

```diff
@@ -30,9 +30,9 @@ func ReadOpensearchClientConfig() (OpensearchClientConfig, error) {
 		Int("opensearch_client_config_port", config.Port).
 		Bool("opensearch_client_config_https", config.Https).
 		Str("opensearch_client_config_auth_username", config.AuthUsername).
-		Str("opensearch_client_config_auth_password", config.AuthPassword).
+		Bool("opensearch_client_config_auth_password_set", config.AuthPassword != "").
```

Comment:
I changed the secret field to denote if the password is set or not, let me know if you are okay with this approach.

Reply:
As Marius convinced me, this is not an issue we should try to solve with logging but with validation. The password field is required, so we can rely on it being set at startup and don't need to verify it in the logs. 
`AuthPassword string `validate:"required" viperEnv:"ELASTIC_AUTH_PASS"``
I would just remove this line then.
```suggestion
```